### DISCUSSION TITLE
Ability to sort Search, Likes, and Mutual Matches by various criteria

### DISCUSSION THREAD

**Discussion by Oshyan** [PullDating/Org#113] [Ideas & Features]
I view Filtering and Sorting, along with open Search functions, as a key differentiator from most current dating apps (which are usually a linear queue these days). When searching, it's important to be able to not just search and filter, but also *sort* your matches in various ways as-needed. We should allow sorting on almost any criteria that is in a user's profile, if possible. Including, but not limited to:
* Match percentage
* Location/distance
* Number of questions answered
* Profile length/completeness
* Date of last sign-on to the app
* Number of photos
* Possibly physical attributes, e.g. height (as a man I actually would value this too, not just for women looking for all guys 😄)

Depending on whether we implement #85 (limit concurrent matches), it may also be useful to be able to sort one's visible Like queue (who Liked you), and possibly the mutual matches as well. If there are more than the proposed limit of 5, it would be nice to have stats info be available in that matching screen, and to be able to sort on it. After all if you have multiple matches, you want some way to decide and prioritize between them. If you're on the fence between 2-3 people, a reminder that "Oh right, this person is actually 15mi from me, whereas this other person is 5 mi" or "This guy is only a 65% match, I should look at what questions we disagree on", etc. could be useful even post-match.

**Comment by ajbott** [comment]
100% agree. This feature should be integrated into the matching/who likes you/matching interfaces of the app. The app also help you with other various statistic in determining who is the best match for you. This could be a build-in app statistic of match %, personality match %, vibe match, profile quality, ect. This could also be in app statistics such as how many filters they have, profile views, current # of matches, how many relationships they've been in, when they joined, new user, ect.

**Comment by viktor-lesyk** [comment]
@ajbott @WilliamDormer do we have a matrix of fields (properties), that we plan to gather about each person, as well as each relation (Person A <> Person B)

I believe that should help us indicate which properties will be searchable, and used for filtering and/or sorting.

**Reply by WilliamDormer** [reply]
I agree we should put something like this together, because it's clear we're going to have some sort of filtering feature off the bat, although not sure how complex we're going to get with it. Oshyan's got some good ideas in there, plus there are the default ones that all of the dating apps have. The only problem is that to make things filterable with relational databases, we're going to need a row in the table for each user for each filter, which adds some complexity if we go too crazy with it. 

But at least to start, we should come up with some "ideal" filtering characteristics, ideally based on some kind of research, or at least consensus from a more diverse crowd (e.g. women too lol)
